docs/developer-guide.md

```diff
@@ -199,6 +199,60 @@ cd java
 mvn clean compile checkstyle:check spotbugs:check
 ```
 
+### Testing
+
+The Maven project includes unit tests, integration tests and system tests. We use JUnit 5, with AssertJ for assertions.
+We also have a setup for manual testing against a deployed instance of Sleeper, documented in
+the [system tests guide](development/system-tests.md#manual-testing).
+
+A unit test is any test that runs entirely in-memory without any I/O operations (eg. file system or network calls).
+If you configure your IDE to run all unit tests at once, they should finish in less than a minute. The unit of a test
+should be a particular behaviour or scenario, rather than eg. a specific method.
+
+A system test is a test that works with a deployed instance of Sleeper. These can be found in the
```

Comment:
To be clearer say "... that works with an instance of Sleeper deployed on AWS." ? (I assume if we're using LocalStack that doesn't count as a system test.)